docs/source/extend_kedro/architecture_overview.md

```diff
@@ -0,0 +1,48 @@
+# Kedro architecture overview
+
+![Kedro architecture diagram](../meta/images/kedro_architecture.png)
+
+At a high level, Kedro consists of five main parts:
+
+## Kedro project
+
+As a data pipeline developer, you will interact with a Kedro project, which consists of:
+
+* The **`conf/`** directory, which contains configuration for the project, such as data catalog configuration, parameters, etc.
+* The **`src`** directory, which contains the source code for the project, including:
+  * The **`pipelines`**  directory, which contains the source code for your pipelines.
+  * **`settings.py`** file contains the settings for the project, such as library component registration, custom hooks registration, etc. All the available settings are listed and explained in the [project settings chapter](../kedro_project_setup/settings.md).
+  * **`pipeline_registry.py`** file defines the project pipelines, i.e. pipelines that can be run using `kedro run --pipeline`.
+  * **`__main__.py`** file serves as the main entry point of the project in [package mode](../tutorial/package_a_project.md#package-a-kedro-project).
+* **`pyproject.toml`** identifies the project root by providing project metadata, including:
+  * `package_name`: A valid Python package name for your project package.
+  * `project_name`: A human readable name for your project.
+  * `kedro_init_version`: Kedro version with which the project was generated.
+
+## Kedro starter
+
+You can use a [Kedro starter](../kedro_project_setup/starters.md) to generate a Kedro project that contains boilerplate code. We maintain a set of [official starters](https://github.com/kedro-org/kedro-starters/) but you can also use a custom starter of your choice.
+
+## Kedro library
+
+Kedro library consists of independent units, each responsible for one aspect of computation in a data pipeline:
+
+* **`Config Loader`** provides utility to parse and load configuration defined in a Kedro project.
+* **`Pipeline`** provides a collection of abstractions to model data pipelines.
+* **`Runner`** provides an abstraction for different execution strategy of a data pipeline.
+* **`I/O`** provides a collection of abstractions to handle I/O in a project, including `DataCatalog` and many `Dataset` implementations.
```

Comment:
Should we add links to the reference documentation, or other places in the documentation that describe these topics?